components/editor/modules/embed/EmbedLoader.js

```diff
@@ -70,6 +70,10 @@ export default (query, Component) => (
       const active = editor.value.blocks.some(
         block => block.key === node.key
       )
+
+      const data = node.data.toJS()
+      // We need to guard against undefined thumbnail below which happens
```

Comment:
let's fix the root issue: the loading state is broken. Which is also responsible for quick flashes of NaN dated tweets in our demo.

In the constructor `loading: false` is bs—should be `loading: !node.data.has('id'), error: !node.data.has('id') && !node.data.has('queryParams') && 'No embed params found.'` (props and therefore node is available as the first arg of the constructor)

Reply:
Done as just paired.